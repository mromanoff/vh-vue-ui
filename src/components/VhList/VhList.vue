<template>
  <component
    :is="as"
    class="VhList"
    :class="computedClasses"
  >
    <slot />
  </component>
</template>

<script>
export default {
  name: 'VhList',
  props: {
    as: {
      type: String,
      default: 'div',
      validator: value => {
        return value.match(/(div|ul|ol|a)/);
      },
    },
    horizontal: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    computedClasses() {
      return {
        'VhList--horizontal': this.horizontal,
      };
    },
  },
};
</script>

<style scoped>
/* BLOCK */

.VhList {
  display: flex;
  flex-direction: column;

  & >>> .VhListItem:not(:last-child) {
    margin-bottom: var(--space--medium);
  }
}

/* MODIFIERS */

.VhList--horizontal {
  flex-direction: row;

  & >>> .VhListItem:not(:last-child) {
    margin-bottom: 0;
    margin-right: var(--space--medium);
  }
}
</style>
