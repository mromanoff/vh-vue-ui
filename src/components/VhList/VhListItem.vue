<template>
  <component
    :is="as"
    class="VhListItem"
    v-on="$listeners"
  >
    <div class="VhListItem-icon">
      <vh-icon
        v-if="icon"
        :name="icon"
      />
    </div>
    <slot />
    <div class="VhListItem-rightContent">
      <slot name="right-content" />
    </div>
  </component>
</template>

<script>
export default {
  name: 'VhListItem',
  props: {
    as: {
      type: String,
      default: 'div',
      validator: value => {
        return value.match(/(div|li|nav)/);
      },
    },
    icon: {
      type: String,
      default: null,
    },
  },
};
</script>

<style scoped>
.VhListItem {
  display: flex;
  align-items: center;
}

.VhListItem-icon {
  margin-right: var(--space--small);
}

.VhListItem-rightContent {
  margin-left: auto;
}

.VhListItem-icon:empty,
.VhListItem-rightContent:empty {
  display: none;
}
</style>
