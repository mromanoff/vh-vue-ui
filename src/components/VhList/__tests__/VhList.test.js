import { shallowMount } from '@vue/test-utils';
import VhList from '../VhList.vue';

describe('VhList.vue', () => {
  let wrapper;

  it('renders default VhList', () => {
    wrapper = shallowMount(VhList);
    expect(wrapper.props().as).toBe('div');
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhList ul', () => {
    wrapper = shallowMount(VhList, {
      propsData: {
        as: 'ul',
      },
    });
    expect(wrapper.props().as).toBe('ul');
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhList ol ', () => {
    wrapper = shallowMount(VhList, {
      propsData: {
        as: 'ol',
        horizontal: true,
      },
    });
    expect(wrapper.props().as).toBe('ol');
    expect(wrapper.props().horizontal).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhList a', () => {
    wrapper = shallowMount(VhList, {
      propsData: {
        as: 'a',
      },
    });
    expect(wrapper.props().as).toBe('a');
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });
});
