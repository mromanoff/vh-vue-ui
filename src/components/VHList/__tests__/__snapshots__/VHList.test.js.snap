// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHList.vue renders default VHList 1`] = `<div class="VHList"></div>`;

exports[`VHList.vue renders modified VHList a 1`] = `<a class="VHList"></a>`;

exports[`VHList.vue renders modified VHList ol  1`] = `<ol class="VHList VHList--horizontal"></ol>`;

exports[`VHList.vue renders modified VHList ul 1`] = `<ul class="VHList"></ul>`;
